tools/policy-gen/crd-extract-openapi.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+POLICY=$1
+VERSION=${2:-"v1alpha1"}
+
+POLICIES_DIR=pkg/plugins/policies
+POLICIES_API_DIR="${POLICIES_DIR}/${POLICY}/api/${VERSION}"
+POLICIES_CRD_DIR="${POLICIES_DIR}/${POLICY}/k8s/crd"
+
+SCHEMA_TEMPLATE=tools/policy-gen/templates/schema.yaml
+
+# 1. Copy file ${SCHEMA_TEMPLATE} to ${POLICIES_API_DIR}/schema.yaml. It contains
+#    information about fields that are equal for all resources 'type', 'mesh' and 'name'.
+#
+# 2. Using yq extract item from the list '.spec.version[]' that has ${VERSION} and
+#    take '.schema.openAPIV3Schema.properties.spec'.
+#
+# 3. Delete 'type' and 'description' for the extracted item, because these are 'type'
+#    and 'description' for the 'spec' field.
+#
+# 4. Using yq eval-all with ireduce merge the file from Step 1 and output from Step 3,
+#    placing the result into the file from Step 1
+
+echo "Generating schema for ${POLICY}/${VERSION} based on CRD"
+
+function cleanupOnError() {
+    rm "${POLICIES_API_DIR}"/schema.yaml
+    echo "Script failed, schema.yaml wasn't generated"
+}
+trap cleanupOnError ERR
+
+cp "${SCHEMA_TEMPLATE}" "${POLICIES_API_DIR}"/schema.yaml
+
+if [ "$(find "${POLICIES_CRD_DIR}" -type f | wc -l | xargs echo)" != 1 ]; then
+  echo "Exactly 1 file is expected in ${POLICIES_CRD_DIR}"
+  exit 1
+fi
+
+CRD_FILE=$(find "${POLICIES_CRD_DIR}" -type f)
+
+# we don't want expressions to be expanded with yq, that's why we're intentionally using single quotes
+# shellcheck disable=SC2016
+yq e '.spec.versions[] | select (.name == "'"${VERSION}"'") | .schema.openAPIV3Schema.properties.spec | del(.type) | del(.description)' \
```

Comment:
I tried the option you proposed, but for some reason, it doesn't work. It just ignores escaping (debug mode output):
```
+ yq e '.spec.versions[] | select (.name == \"${VERSION}\") | .schema.openAPIV3Schema.properties.spec | del(.type) | del(.description)' pkg/plugins/policies/donothingpolicy/k8s/crd/kuma.io_donothingpolicies.yaml
+ yq eval-all -i '. as $item ireduce ({}; . * $item )' pkg/plugins/policies/donothingpolicy/api/v1alpha1/schema.yaml -
Error: parsing expression: Lexer error: could not match text starting at 1:37 failing at 1:38.
        unmatched text: "\\"
```

I can leave a comment about why we need `"'"`:
* the first double quote is needed because `yq select` expects something like `select (.name == "v1alpha1")`
* single quote is required to close the first single quote
* now when we outside of single quotes we simply use bash expression `"${VERSION}"`, that's why the last double quote is here